src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -636,43 +658,71 @@ private bool StartCore(ProcessStartInfo startInfo)
 
                     if (startInfo.UserName.Length != 0)
                     {
+                        if(startInfo.Password != null && startInfo.PasswordInClearText != null)
+                        {
+                            throw new ArgumentException(SR.CantSetDuplicatePassword);
+                        }
+
                         Interop.mincore.LogonFlags logonFlags = (Interop.mincore.LogonFlags)0;
                         if (startInfo.LoadUserProfile)
                         {
                             logonFlags = Interop.mincore.LogonFlags.LOGON_WITH_PROFILE;
                         }
 
-
-                        try { }
-                        finally
+                        string password = string.Empty;
+                        IntPtr secureStringPtr = IntPtr.Zero;
+                        try
                         {
-                            retVal = Interop.mincore.CreateProcessWithLogonW(
-                                    startInfo.UserName,
-                                    startInfo.Domain,
-                                    startInfo.PasswordInClearText,
-                                    logonFlags,
-                                    null,            // we don't need this since all the info is in commandLine
-                                    commandLine,
-                                    creationFlags,
-                                    environmentPtr,
-                                    workingDirectory,
-                                    startupInfo,        // pointer to STARTUPINFO
-                                    processInfo         // pointer to PROCESS_INFORMATION
-                                );
+                            if(startInfo.Password != null)
+                            {
+                                secureStringPtr = Marshal.SecureStringToGlobalAllocUnicode(startInfo.Password);
+                                password = Marshal.PtrToStringUni(secureStringPtr);
```

Comment:
The [desktop code](http://index/?query=GetProcessesByName&rightProject=System&file=%5BRepoRoot%5D%5CNDP%5Cfx%5Csrc%5Cservices%5Cmonitoring%5Csystem%5Cdiagnosticts%5CProcess.cs&line=2076) avoids creating a regular string from the securestring. It passes the pointer to it to CreateProcess. Can we do that?